### PR TITLE
ARROW-1976: [Python] Handling unicode pandas columns on parquet.read_table

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: python/pyarrow/pandas_compat.py [hunk 1]
```diff
@@ -172,7 +172,10 @@ def get_column_metadata(column, name, arrow_type, field_name):
 
     return {
         'name': name,
-        'field_name': str(field_name),
+        'field_name': (
+            'None' if field_name is None
+            else _column_name_to_strings(field_name)
```
Comment (jorisvandenbossche) [inline python/pyarrow/pandas_compat.py]: Is this call to `_column_name_to_strings` necessary? I would think that was already done for field_name when you end up here

**Comment by wesm** [comment]
Any chance anyone has time to look at getting our CI rehabilitated so we can get a passing CI build? https://github.com/apache/arrow/pull/1548 still has some Windows issues

**Comment by cpcloud** [comment]
@wesm Looking now.

File: python/pyarrow/pandas_compat.py (hunk 1 above)
Comment (cpcloud) [reply python/pyarrow/pandas_compat.py]: Nope, removed it.

**Review by wesm — APPROVED** [review]
+1, thanks for helping with this @cpcloud! I fixed a flake and rebased so the build should pass now

**Comment by wesm** [comment]
Appveyor build is running here: https://ci.appveyor.com/project/cpcloud/arrow/build/1.0.516. Will merge once that is looking good

**Comment by simnyatsanga** [comment]
@wesm @cpcloud Thanks so much for resolving this issue!

**Pull request closed without merging**